ng-app/app/views/task.html

```diff
@@ -30,48 +30,63 @@ <h2 class="md-flex">{{ task.displayName }} {{ task.name }}</h2>
   </div>
   <!-- /Task description -->
   <!-- Comment description -->
-  <div layout="column" class="line-vertical">
-    <md-whiteframe class="md-whiteframe-z1" layout-margin ng-repeat="comment in task.comments">
-      <span>
-      <i class="mdi mdi-checkbox-multiple-blank-outline task-activity-icon button-icon-font"></i>
-      </span>
-      <md-toolbar class="comment-container">
-        <div class="md-toolbar-tools md-subhead comment-container">
-          <div layout-margin>
-            <img gravatar-src="comment.updatedBy.email" class="md-avatar img-circle vertical-align-middle" gravatar-size="38" alt="{{ comment.updatedBy.displayName }}"
-              ng-show="comment.objectType == 'project.work.history'"/>
+  <div class="line-vertical">
+    <div layout="column" class="comment-box" ng-repeat="comment in task.comments" layout-margin>
+      
+      <md-whiteframe class="md-whiteframe-z1" ng-if="comment.objectType == 'project.work.history'">
+        <md-toolbar class="comment-container">
+          <div class="md-toolbar-tools md-subhead comment-container">
+            <div layout-margin>
+              <img gravatar-src="comment.updatedBy.email" class="md-avatar img-circle vertical-align-middle" gravatar-size="38" alt="{{ comment.updatedBy.displayName }}"/>
+            </div>
+            <div class="text-black toolbar-font">
+              {{ comment.updatedBy.displayName }} commented on {{ comment.create_date | date:'medium'}}
+            </div>
+             <span flex></span>
           </div>
-          <div class="text-black toolbar-font">
-            {{ comment.updatedBy.displayName }} commented on {{ comment.create_date | date:'medium'}}
+        </md-toolbar>
+        <div layout-padding layout-margin>
+          <div class="text-black">
+            <p marked="comment.comment"><p>
           </div>
-           <span flex></span>
+        </div>
+      </md-whiteframe>
+
+      <div class="row activity-box">
+        <div class="text-black" ng-if="comment.new_assignee">
+          <i class="mdi mdi-checkbox-multiple-blank-outline task-activity-icon"></i>
+          Assigned to {{ comment.new_assignee.displayName }}
+        </div>
+        <div class="text-black task-state" ng-if="comment.objectType == 'timesheet.line'">
+          <i class="mdi mdi-checkbox-multiple-blank-outline task-activity-icon"></i>
+          Worked for {{ comment.displayName }}
+        </div>
+        <div class="text-black" ng-if="comment.new_progress_state">
+          <i class="mdi mdi-checkbox-multiple-blank-outline task-activity-icon"></i>
+          Changed task state from 
           <span class="progress-change" ng-if="comment.new_progress_state">
-            <span>{{ comment.previous_progress_state }}</span>
-            <i class="mdi mdi-arrow-right-bold"></i>
-            <span>{{ comment.new_progress_state }}</span>
+            <span class="text-black">{{ comment.previous_progress_state }}</span>
```

Comment:
Why do you need text-black in this? If you must add it to all children, might as well just put it in the parent element.